starter-code/js/index.js

```diff
@@ -1,52 +1,104 @@
+//   itemQty = Number(itemQty);
 
-}
+//   // calculates the total price based on this data
+//   var productTotalPrice = itemPrice * itemQty;
 
-function createItemNode(dataType, itemData){
+//   // Updates the total price in the DOM
+//   var singleItemSum = document.querySelector(".single-item-sum");
+//   singleItemSum.innerHTML = "$" + productTotalPrice;
+// };
 
-}
+// selecting all the items and creating an array
+var itemsArray = document.querySelectorAll(".item");
 
-function createNewItemRow(itemName, itemUnitPrice){
+/* LA FUNCTION POUR UN PANIER A PLUSIEURS LIGNES (ITEMS) EN LOOP */
 
-}
+// when user click on the green button calculate
+calculatePriceButton.onclick = function() {
+  // initial total sum of the cart
+  var cartTotalSum = 0;
 
-function createNewItem(){
+  //for each item row in the cart (pour chaque ligne du panier)
+  itemsArray.forEach(function(oneRow, i) {
+    console.log("item n#" + i + "contenu html" + oneRow.innerHTML);
 
-}
+    var itemPrice = oneRow.querySelector(".item-price").innerHTML;
+    itemPrice = parseFloat(itemPrice);
 
-window.onload = function(){
-  var calculatePriceButton = document.getElementById('calc-prices-button');
-  var createItemButton = document.getElementById('new-item-create');
-  var deleteButtons = document.getElementsByClassName('btn-delete');
+    // creating a dynamic css #id selector for each inpu (see html code)
+    var itemId = "#item" + i + "-qty";
 
-  calculatePriceButton.onclick = getTotalPrice;
-  createItemButton.onclick = createNewItem;
+    // retrieves the quantity of items desired by fetching value propierty of EACH input
+    var itemQty = document.querySelector(itemId).value;
+    itemQty = Number(itemQty);
 
-  for(var i = 0; i<deleteButtons.length ; i++){
-    deleteButtons[i].onclick = deleteItem;
-  }
+    // calculates the total price based on this data (for THIS row)
+    var productTotalPrice = itemPrice * itemQty;
+
+    // Updates the total price in the HTML (according to the number of desired items)
+    var singleItemSum = oneRow.querySelector(".single-item-sum");
+
+    // adding a dollar sign (becaaauuse)
+    singleItemSum.innerHTML = "$" + productTotalPrice;
+
+    // add it to the cart's total sum (initially 0)
+    cartTotalSum += productTotalPrice;
+  }); // end of my loop
+
+  // getting total cart sum
+  console.log(cartTotalSum);
+
+  var cartTotalResult = document.querySelector(".cart-total-result");
+
+  //calculating cart total price
+  cartTotalResult.innerHTML = "$" + cartTotalSum;
 };
+
+/*
+Iteration 3: Calculating the total price of all products
+
+Now that you have each product's total price, you need to calculate the total price of the entire shopping cart.
+Once you have that number, you need to display the result in the DOM.
+*/
+//calculatePriceButton.onclick = getTotalPrice;
+//createItemButton.onclick = createNewItem;
+
+for (var i = 0; i < deleteButtons.length; i++) {
+  deleteButtons[i].onclick = deleteItem;
+}
```

Comment:
It was a good beginning. That's kind of the method. Just a part is missing. 

Here is the solution:
```
// Defining the function, will be called upon clicking on a button
  deleteItem = function( event ) {

    // event here symbolizes the click
    // event.target selects the element the click happened on, allowing us to find exactly which button was clicked
    // parentNode selects the current node's direct parent
    // remove() actually removes the DOM node from the DOM, deleting the line on the page
    event.target.parentNode.parentNode.remove();
  }

  // Saving the delete buttons inside a collection of buttons, so I can apply my onclick on all of them
  var deleteButtons = document.querySelectorAll(".btn-delete");

  // Looping over all the buttons of my collection
  deleteButtons.forEach( function(each) {

    // Applying the same onclick function on all of them, so any of them would trigger the deleteItem function when clicked
    each.onclick = deleteItem;
  });
```